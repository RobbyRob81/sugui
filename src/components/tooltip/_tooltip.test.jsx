import * as React from 'react';
import { customRender } from '../../../test/test-utils';
import Tooltip from '.';
import Button from '../button';

class TooltipExample extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: false
    };

    this.ref = null;
  }

  render() {
    return (
      <div ref={(c) => { this.ref = c; }} style={{ width: '130px' }}>
        <Button
          color="primary"
          onMouseEnter={() => this.setState({ value: true })}
          onMouseLeave={() => this.setState({ value: false })}
        >
          Show Tooltip
        </Button>
        {this.state.value ?
          <Tooltip parent={this.ref}>
            <span>This is just an example tooltip</span>
          </Tooltip>
        : null}
      </div>
    );
  }
}

it('should render without throwing an error', () => {
  const { container } = customRender(<TooltipExample />);
  expect(container.firstChild).toMatchSnapshot();
});
