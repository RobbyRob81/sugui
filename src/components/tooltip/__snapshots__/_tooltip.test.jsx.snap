// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `
<div
  style="width: 130px;"
>
  <button
    class="sc-bdVaJa bNuAbZ"
    color="primary"
  >
    Show Tooltip
  </button>
</div>
`;
